appshell/update_mac.mm

```diff
@@ -172,9 +173,62 @@ + (void) RunAppUpdate{
+                        NSAppleEventDescriptor *returnDescriptor = NULL;
+                        NSString * appleScriptString = @"do shell script \"";
+                        appleScriptString = [appleScriptString stringByAppendingString:nohupPath];
+                        appleScriptString = [appleScriptString stringByAppendingString:@" "];
+                        appleScriptString = [appleScriptString stringByAppendingString:shPath];
+                        appleScriptString = [appleScriptString stringByAppendingString:@" "];
+                        appleScriptString = [appleScriptString stringByAppendingString:scriptPath];
+                        appleScriptString = [appleScriptString stringByAppendingString:@" -a "];
+                        appleScriptString = [appleScriptString stringByAppendingString:bracketsAppName];
+                        appleScriptString = [appleScriptString stringByAppendingString:@" -b "];
+                        appleScriptString = [appleScriptString stringByAppendingString:@"'"];
+                        appleScriptString = [appleScriptString stringByAppendingString:installDir];
+                        appleScriptString = [appleScriptString stringByAppendingString:@"'"];
+                        appleScriptString = [appleScriptString stringByAppendingString:@" -l "];
+                        appleScriptString = [appleScriptString stringByAppendingString:@"'"];
+                        appleScriptString = [appleScriptString stringByAppendingString:logFilePath];
+                        appleScriptString = [appleScriptString stringByAppendingString:@"'"];
+                        appleScriptString = [appleScriptString stringByAppendingString:@" -m "];
+                        appleScriptString = [appleScriptString stringByAppendingString:mountPoint];
+                        appleScriptString = [appleScriptString stringByAppendingString:@" -t "];
+                        appleScriptString = [appleScriptString stringByAppendingString:@"'"];
+                        appleScriptString = [appleScriptString stringByAppendingString:updateDir];
+                        appleScriptString = [appleScriptString stringByAppendingString:@"'"];
+                        appleScriptString = [appleScriptString stringByAppendingString:@" -p "];
+                        appleScriptString = [appleScriptString stringByAppendingString:pidString];
+                        appleScriptString = [appleScriptString stringByAppendingString:@" > /dev/null 2>&1 &"];
+                        appleScriptString = [appleScriptString stringByAppendingString:@"\""];
+                        
+                        if(!isAdmin) {
+                            appleScriptString = [appleScriptString stringByAppendingString:@" with administrator privileges"];
+                        }
+
+                        NSAppleScript *theScript = [[NSAppleScript alloc] initWithSource:appleScriptString];
```

Comment:
`theScript` string needs be freed. But anyways since this in the `destruction` sequence I guess we can live with it.